crates/sqlexec/src/remote/client.rs

```diff
@@ -193,14 +180,33 @@ impl RemoteClient {
         }
 
         let mut dst: Endpoint = dst.try_into()?;
-        if let Some(tls_conf) = tls_conf {
-            metadata.insert(CA_CERT_PATH, tls_conf.ca_cert_path.parse()?);
-            metadata.insert(DOMAIN, tls_conf.domain.parse()?);
-            let ca = std::fs::read_to_string(tls_conf.ca_cert_path)?;
+
+        let mut api_url = "https://console.glaredb.com/api/internal/authenticate/client";
+        let host = dst.uri().host().expect("invalid host");
+        if host.contains("qa.glaredb.com") {
+            api_url = "https://qa.glaredb.com/api/internal/authenticate/client";
+        }
+
+        if !disable_tls {
+            let mut body = HashMap::new();
+            body.insert("user", params.user);
+            body.insert("password", params.password);
+            body.insert("org_name", params.org);
+            body.insert("db_name", params.db_name);
+
+            let client = reqwest::Client::new();
+            let res = client
```

Comment:
client creation is expensive (ish) and part of my "can we do this closer to the startup or when we construct the remote client". I'm not sure how re-connections are handled, but 

another option would be to pass the credential config in as a future, if  you didn't want to resolve the certs until you try to connect. This way (I think) subsequent attempts to resolve the future would get an effectively cached value, and you would end up doing the request only once.